src/utils/dex/name.js

```diff
@@ -0,0 +1,84 @@
+import { PersonalTable, basePokemonNames, formPokemonNames } from '../../../__gamedata';
+import { FORM_MAP } from './functions';
+
+const POKEMON_NAME_MAP = PersonalTable.Personal.reduce(createPokemonMap, {});
+
+function createPokemonMap(pokemonNameMap, currentPokemon) {
+  try {
+    const { id } = currentPokemon;
+
+    const baseFormname = basePokemonNames.labelDataArray[id]?.wordDataArray[0]?.str;
+    if (typeof baseFormname === 'string' && baseFormname.length > 0) {
+      pokemonNameMap[id] = baseFormname;
+      return pokemonNameMap;
+    }
+
+    const alternateFormName = formPokemonNames.labelDataArray[id]?.wordDataArray[0]?.str;
+    if (typeof alternateFormName === 'string' && alternateFormName.length > 0) {
+      pokemonNameMap[id] = alternateFormName;
+      return pokemonNameMap;
+    }
+
+    pokemonNameMap[id] = getFormNameOfProblematicPokemon(id);
+    return pokemonNameMap;
+  } catch (e) {
+    throw Error(`${currentPokemon.id} - ${e}`);
+  }
+}
+
+function getFormName(id = 0) {
+  return POKEMON_NAME_MAP[id];
+}
+
+function getPokemonName(pokemonId = 0) {
+  return POKEMON_NAME_MAP[pokemonId];
+}
+
+function getPokemonIdFromName(name = 'Egg') {
+  const id = Object.values(POKEMON_NAME_MAP).findIndex((e) => e === name);
+  return id === -1 ? 0 : id;
```

Comment:
I like to default to dummy data when I can, it reduces the amount of code required, as well as cognitive load handling all the alternatives. We can also be a little lazier about this since we have a controlled dataset, we know what we can expect.

Reply:
alright 👍